src/classes/UTIL_DMLService.cls

```diff
@@ -0,0 +1,306 @@
+        
+        dmlOptions.optAllOrNone = ALL_OR_NONE_FALSE;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordsToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToInsert The sObject to add to the org.
+    * @param allOrNone The optional parameter that specifies whether the operation allows partial success.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult insertRecord(sObject recordToInsert, Boolean allOrNone) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToInsert The sObjects to add to the org.
+    * @param allOrNone The optional parameter that specifies whether the operation allows partial success.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> insertRecords(List<sObject> recordsToInsert, Boolean allOrNone) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordsToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToInsert The sObject to add to the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult insertRecord(sObject recordToInsert, Database.DMLOptions dmlOptions) {
+        dmlOptions.optAllOrNone = ALL_OR_NONE_FALSE;
```

Comment:
if the user passes in dmlOptions, I don't think we should be overriding their choice of optAllOrNone.

Reply:
I agree. In the UTIL_DMLService doc, I'll make sure to call out that along with any dmlOptions a dev might set, they will also need to set the optAllOrNone.